libbeat/template/config.go

```diff
@@ -17,7 +17,27 @@
 
 package template
 
-import "github.com/elastic/beats/v7/libbeat/mapping"
+import (
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/mapping"
+)
+
+const (
+	TemplateLegacy TemplateType = iota
```

Comment:
I suggest using the full name `IndexTemplate` on all "Templates". Now it's obvious, but in a few months for someone who has not been directly involved in this it might be a bit confusing, also when doing a `grep` search. It's also more specific if later we have something like, I don't know, `MappingTemplates`? :smile: